sparkmagic/sparkmagic/controllerwidget/manageendpointwidget.py

```diff
@@ -30,7 +30,13 @@ def get_existing_endpoint_widgets(self):
 
             # Endpoints
             for url, endpoint in self.endpoints.items():
-                endpoint_widgets.append(self.get_endpoint_widget(url, endpoint))
+                try:
+                    endpoint_widgets.append(self.get_endpoint_widget(url, endpoint))
+                except HttpClientException:
+                    # If we can't reach one of the default endpoints, just skip over it
+                    # silently
+                    if not endpoint.is_default:
```

Comment:
Fair enough. Added a log to that effect. I hope `SparkLogger` was the right thing to use here? It seems to just be a simple subclass of the hdijupyterutils Log object, but the name confused me... made me think it somehow logged things on/about the Spark cluster.